### DISCUSSION TITLE
Connection Error

### DISCUSSION THREAD

**Discussion by lirtm** [make-all/tuya-local#735] [General]
Previously here: https://github.com/make-all/tuya-local/discussions/522#discussioncomment-5890770 wrote about a problem that one of the outlets was not being added, swearing about similar errors from that thread. In the last update it was added and worked steadily.
A few days ago it briefly started to switch to "unavailable". As I understand it, only in HA.
After - at night was unavailable for a couple of hours. Now it's happening again.

I see the following in the logs:

Level: WARNING
```
Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:225
Integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: 16:57:39 (1 occurrences)
Last logged: 16:57:39
UPS receive loop has terminated
```

Level: WARNING
```
 Logger: homeassistant.util.async_
Source: util/async_.py:159
First occurred: 21 мая 2023 г. в 23:20:34 (8 occurrences)
Last logged: 17:01:07

    Detected blocking call to sleep inside the event loop. This is causing stability issues. Please report issue to the custom integration author for tuya_local doing blocking calls at custom_components/tuya_local/device.py, line 78: self._api = tinytuya.Device(dev_id, address, local_key)
    Detected blocking call to sleep inside the event loop. This is causing stability issues. Please report issue to the custom integration author for tuya_local doing blocking calls at custom_components/tuya_local/device.py, line 75: parent=tinytuya.Device(dev_id, address, local_key),

```

Level: ERROR
```
Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:202
Integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: 21 мая 2023 г. в 23:21:39 (1411 occurrences)
Last logged: 17:05:38

    Failed to update device dps for UPS
    UPS error reading: Unexpected Payload from Device
    Failed to fetch device status for UPS
```

Any thoughts?

**Comment by lirtm** [comment]
In the end, what do we have?
Bought an Aubess socket, with one socket, which was explicitly listed as supported.
It was added immediately, but as soon as I removed it from the main socket and inserted it elsewhere, the error data appeared again and the socket did not appear. I decided to remove this new socket from the configuration and add it again. And... It appeared in HA. Did the same with the one that is the subject of this thread - and it appeared, values and everything else.
Yes, I have the sockets added in parallel to the standard Tuya integration. Is this just the sort of thing that might be affecting it?

And as I understand it, there is no way to remove the individual outlet from the standard integration? As all the others are working steadily.

**Comment by make-all** [comment]
Parallel to the cloud Tuya integration should not cause connection issues, but may cause some strange behaviour such as toggling on/off immediately after a change if both integrations try to control the device at the same time.  Parallel to other local connections will cause problems.  Even just having localtuya installed prevents Auto IP address detection from working (but that feature is problematic anyway, as it causes warnings to appear in the HA log due to the way the underlying tinytuya library is working).

But I don't quite understand whether you still have an issue, or you resolved it by removing and re-adding the device (which possibly means the local key or IP address changed).

**Reply by lirtm** [reply]
Ah, that is, as I understand it, localtuya is better to uninstall from the system at all, because even its mere presence causes problems?

At this point, I went back to standard integration for one of these outlets because I preferred the stability of the frequency of the values given. But, in principle, I can try again.

But it is interesting that Yandex-sockets that are tied to the standard Tuya integration are not subject to problems when using tuya-local even when localtuya is present, while rosettes that treat the Tuya ecosystem as the primary one have problems in this case.

If there are any problems, I'll add info to the thread.

Thank you!

**Reply by make-all** [reply]
localtuya only causes the Auto IP address to not work, if you do not set up the same device in both integrations.  But it is better to configure the IP address anyway.

**Reply by make-all** [reply]
"that treat the Tuya ecosystem as the primary one" - does this mean you use the Tuya mobile app?  When the mobile device is on WiFi, it will use local connections to the devices, so it will interfere with the tuya-local (or localtuya) connection, since many cheaper Tuya devices support only one local connection (mostly lightbulbs - smart sockets often support up to 3, but some only 1).

**Reply by lirtm** [reply]
> localtuya only causes the Auto IP address to not work, if you do not set up the same device in both integrations. But it is better to configure the IP address anyway.

yee, I most often connected without specifying an IP address. My outlet address itself is static.



> "that treat the Tuya ecosystem as the primary one" - does this mean you use the Tuya mobile app? When the mobile device is on WiFi, it will use local connections to the devices, so it will interfere with the tuya-local (or localtuya) connection, since many cheaper Tuya devices support only one local connection (mostly lightbulbs - smart sockets often support up to 3, but some only 1).

yee. I connect the outlets through the SmartLife app. Sometimes I run this application at home, being connected to the same router as the outlets, but at 5 GHz and a wi-fi network with a different name.
And how can you determine (not experimentally) the number of possible connections? It just turns out that the YNDX-0007W has more possible local connections than the AUBESS 1-socket or the same shape of sockets.

**Comment by lirtm** [comment]
hmm
Uninstalled Local Tuya, restarted, found out with tinytuya local_key.
I enter the data and there is an error. I also did not run the app.
![2023-06-06_192701](https://github.com/make-all/tuya-local/assets/4773960/f7197a81-ebc0-4276-b576-f4d0e83bf2f3)
![2023-06-06_193006](https://github.com/make-all/tuya-local/assets/4773960/d40d9319-1c3b-4a05-ba9b-3fdde913a368)

**Reply by make-all** [reply]
You should only get the error in the last screenshot if you do not specify any IP address (Auto),
If you have other devices, this may be about them rather than the device you are configuring.

Or if the device is already connected, and you are reconfiguring it, then you may see this error, as it is too slow to disconnect before the Test connection starts.
